### PR TITLE
Feature/extended tests

### PR DESCRIPTION
**Pull request by Shea690901**

### CODE REVIEW

**Comment by Shea690901** [comment]
Each commit here will be mergeable on it's own, completing some test in any way.
I'll keep this branch open even after merging and add as I stumble over any problems.

**Comment by Shea690901** [comment]
Oops...
Mistakenly added the wrong changeset...

File: testsuite/single/tests/efuns/replace_string.c [hunk 1]
```diff
@@ -1,10 +1,41 @@
 void do_tests() {
     string src = "This is a test";
 
+    ASSERT(replace_string("aaaaaa", "aa", "x") == "xxx");
```
Comment (thefallentree) [inline testsuite/single/tests/efuns/replace_string.c]: It's better to use ASSERT_EQ , can you fix it?

File: testsuite/single/tests/efuns/dump_prog.c [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+            fn = "/PROG_DUMP_2";
+            break;
+        case 3:
+            fn = "/PROG_DUMP_3";
+            break;
+        default:
+            dump_prog(this_object());
+            return;
+    }
+    dump_prog(this_object(), flag, fn);
```
Comment (thefallentree) [inline testsuite/single/tests/efuns/dump_prog.c]: why do we need a switch but not the same style as the test for replace_string ? I feel it is much easier to read that way, just repeat a few lines that call dump_program() with different flags and arguments. 

Also, if we can't verify the content in anyway , at least we could verify the file exists and non-empty after calling this efun.

**Pull request closed without merging**